contrib/database/sql/option.go

```diff
@@ -88,6 +89,16 @@ func WithChildSpansOnly() Option {
 	}
 }
 
+
+// WithErrorCheck specifies a function fn which determines whether the passed
+// error should be marked as an error. The fn is called whenever a database/sql operation
+// finishes with an error
+func WithErrorCheck(fn func(err error) bool) Option {
```

Comment:
I'm not a huge fan of the naming of this API. `WithErrorCheck` doesn't mean much to me when I look at it. My first question when I looked at the name was whether `fn` returning `true` means that this error should be considered passing or failing.

Perhaps just renaming it to `WithIsErrorFunc` would be enough to make this clearer.

Edit: Just realized this came from https://github.com/DataDog/dd-trace-go/pull/806/files#diff-fb36e8062180ee0c1ec2507fcbf7045a3e023eed177d67e36261cb86af5e647f. Probably better to be consistent, since I imagine we can't change the other one. Nevermind 😢

Reply:
I'm agree, `WithIsErrorFunc` is more clear rather than `WIthErrorCheck`. I entrust to datadog team about this name. @katiehockman @gbbr

Reply:
I agree the naming isn’t great but we’ll have to stay consistent with all the other integrations which use the exact same name for this option.